test/e2e/user_certs_test.go

```diff
@@ -205,7 +210,15 @@ func TestNamedCertificates(t *testing.T) {
 	}
 
 	// wait until a new version has been rolled out with the new configuration
-	testlibraryapi.WaitForAPIServerToStabilizeOnTheSameRevision(t, kubeClient.Pods(operatorclient.TargetNamespace))
+	for {
+		testlibraryapi.WaitForAPIServerToStabilizeOnTheSameRevision(t, kubeClient.Pods(operatorclient.TargetNamespace))
```

Comment:
do we know why having this method is not enough?

Reply:
When we call it the second time, sometimes the rollout of the new configuration has not yet "started" and we are still "synced" on previous revision.

Reply:
the intention of `WaitForAPIServerToStabilizeOnTheSameRevision` method was to wait some time, even if nothing has changed before returning. I think this is the use case we have here.

Reply:
> the intention of `WaitForAPIServerToStabilizeOnTheSameRevision` method was to wait some time, even if nothing has changed before returning. I think this is the use case we have here.

We were still missing the rollout of the configuration changes sometimes if the rollout was slow enough. What I really want is for my config changes to be rolled out, which means at least the next revision. Simply checking for the next revision isn't perfect either, but seems good enough given that all the e2e-operator tests are run serially (i.e. all in same package, no calls to `t.Parallel()`).

As an added bonus, when things are running smoothly, the `e2e-operator` tests spends less time waiting. I don't care past the next 'stable' revision as the tests are running serially.

Reply:
> We were still missing the rollout of the configuration changes sometimes if the rollout was slow enough

I have just checked and the `library.WaitForPodsToStabilizeOnTheSameRevision` waits 6 minutes to ensure the APIs are stable and on the same revision.

Reply:
I have also written a unit test to ensure it can handle a single pod and it seems to be working.

Reply:
are you sure the issue is with the `WaitForPodsToStabilizeOnTheSameRevision` method?

Reply:
ptal, https://github.com/openshift/library-go/pull/1350